lib/segment/src/index/field_index/full_text_index/posting_list.rs

```diff
@@ -37,23 +40,309 @@ impl PostingList {
         self.list.binary_search(val).is_ok()
     }
 
-    pub fn iter<'a>(
-        &'a self,
-        filter: impl Fn(PointOffsetType) -> bool + 'a,
-    ) -> impl Iterator<Item = PointOffsetType> + 'a {
-        self.list.iter().copied().filter(move |&idx| filter(idx))
+    pub fn iter(&self) -> impl Iterator<Item = PointOffsetType> + '_ {
+        self.list.iter().copied()
     }
+}
+
+#[derive(Clone, Debug, Default)]
+pub struct CompressedPostingList {
+    len: usize,
+    last_doc_id: PointOffsetType,
+    data: Box<[u8]>,
+    chunks: Box<[CompressedPostingChunk]>,
```

Comment:
Can't we use a vector in these two cases instead, rather than a boxed array?

Reply:
I placed boxed array to show that we cannot change size. But in general I don't have any doubts. Fixed